lxd/main_forkproxy.go

```diff
@@ -215,7 +220,58 @@ func (c *cmdForkproxy) Run(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("Failed to re-assemble listener: %v", err)
 	}
 
-	defer listener.Close()
+	// Handle SIGTERM which is sent when the proxy is to be removed
+	terminate := false
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM)
+
+	// Wait for SIGTERM and close the listener in order to exit the loop below
+	go func() {
+		<-sigs
+		terminate = true
+		listener.Close()
+	}()
+
+	cAddr := parseAddr(connectAddr)
+	lAddr := parseAddr(listenAddr)
+
+	if cAddr.connType == "unix" && !cAddr.abstract {
+		// Since we prefix the connect address with the container's rootfs path,
+		// we can easily run into the 108 byte limit of the sun_path field in
+		// the corresponding c struct.
+		// To work around this, we chdir into the rootfs, and then create the
+		// unix socket. Afterwards, we return to the old directory.
```

Comment:
Isn't that a disaster waiting to happen because of symlinks/bind-mounts/...?

We should attach to the container's netns, mntns and userns, drop priv to 0/0 inside the container and then do the bind/connect from there so that we both properly handle symlinks inside the container, we respect any bind-mount in place and we don't do the bind/connect with elevated privileges (real root).

@brauner did I miss something?